File: Agency/backend/agency/src/main/java/com/agency/backend/service/interfaces/ElasticsearchService.java
package com.agency.backend.service.interfaces;

import com.agency.backend.dto.AdvancedQueryDto;
import com.agency.backend.dto.GeospatialSearchDto;
import com.agency.backend.dto.SearchResult;
import com.agency.backend.dto.SimpleQueryDto;

import java.io.IOException;
import java.util.List;

public interface ElasticsearchService {

    List<SearchResult> searchByField(SimpleQueryDto queryDto);

    List<SearchResult> searchByFields(List<SimpleQueryDto> queryDto);

    List<SearchResult> searchByCvContent(String cvContent);

    List<SearchResult> searchByCoverLetterContent(String coverLetterContent);

    List<SearchResult> searchByPhrase(SimpleQueryDto simpleQueryDto);

    List<SearchResult> searchByBooleanQuery(AdvancedQueryDto advancedQueryDto);

    List<SearchResult> geospatialSearch(GeospatialSearchDto geospatialSearchDto);
}

File: Agency/frontend/agency-app/src/service/ElasticsearchService.js
import { searchByField, searchByFields, searchByPhrase, booleanSearch, geospatialSearch, searchByCVContent, searchByCoverLetterContent } from "../api/ElasticsearchApi";

export const ElasticsearchService = {

    searchByField: async function(simpleQueryDto){
        return await searchByField(simpleQueryDto);
    },

    searchByFields: async function(firstQuery, secondQuery){
        const simpleQueryList = [];
        simpleQueryList.push(firstQuery);
        simpleQueryList.push(secondQuery);
        return await searchByFields(simpleQueryList);
    },

    searchByCVContent: async function(cvContent) {
        return await searchByCVContent(cvContent);
    },

    searchByCoverLetterContent: async function(coverLetterContent) {
        return await searchByCoverLetterContent(coverLetterContent);
    },

    searchByPhrase: async function(simpleQueryDto){
        return await searchByPhrase(simpleQueryDto);
    },

    booleanSearch: async function(firstQuery, secondQuery, booleanOperation) {
        const advancedSearchQuery = {
            "field1" : firstQuery.field,
            "value1" : firstQuery.value,
            "field2" : secondQuery.field,
            "value2" : secondQuery.value,
            "operation" : booleanOperation
        }
        return await booleanSearch(advancedSearchQuery);
    },

    geospatialSearch: async function(geospatialFormData) {
        return await geospatialSearch(geospatialFormData);
    }
}